pkg/compact/compact_test.go

```diff
@@ -174,3 +179,622 @@ func BenchmarkGatherNoCompactionMarkFilter_Filter(b *testing.B) {
+							Resolution: downsample.ResLevel2,
+						},
+					},
+				},
+			},
+			expected: map[string]float64{
+				keys[0]: 0.0,
+				keys[1]: 0.0,
+				keys[2]: 0.0,
+			},
+		}, {
+			testName: "third_test",
+			input: []*metadata.Meta{
+				{
+					BlockMeta: tsdb.BlockMeta{
+						ULID:    ulid.MustNew(9, nil),
+						MinTime: 0,
+						MaxTime: downsample.DownsampleRange0,
+						Compaction: tsdb.BlockMetaCompaction{
+							Sources: []ulid.ULID{ulid.MustNew(10, nil), ulid.MustNew(11, nil)},
+						},
+					},
+					Thanos: metadata.Thanos{
+						Version: 1,
+						Labels:  map[string]string{"a": "1"},
+						Downsample: metadata.ThanosDownsample{
+							Resolution: downsample.ResLevel0,
+						},
+					},
+				},
+			},
+			expected: map[string]float64{
+				keys[0]: 1.0,
+				keys[1]: 0.0,
+				keys[2]: 0.0,
+			},
+		}, {
+			testName: "fourth_test",
+			input: []*metadata.Meta{
+				{
+					BlockMeta: tsdb.BlockMeta{
+						ULID:    ulid.MustNew(10, nil),
+						MinTime: 0,
+						MaxTime: downsample.DownsampleRange1,
+						Compaction: tsdb.BlockMetaCompaction{
+							Sources: []ulid.ULID{ulid.MustNew(11, nil), ulid.MustNew(12, nil)},
+						},
+					},
+					Thanos: metadata.Thanos{
+						Version: 1,
+						Labels:  map[string]string{"a": "1"},
+						Downsample: metadata.ThanosDownsample{
+							Resolution: downsample.ResLevel1,
+						},
+					},
+				},
+			},
+			expected: map[string]float64{
+				keys[0]: 0.0,
+				keys[1]: 0.0,
+				keys[2]: 0.0,
+			},
+		},
+		{
+			testName: "fifth_test",
+			input: []*metadata.Meta{
+				{
+					BlockMeta: tsdb.BlockMeta{
+						ULID:    ulid.MustNew(0, nil),
+						MinTime: 0,
+						MaxTime: downsample.DownsampleRange1,
+						Compaction: tsdb.BlockMetaCompaction{
+							Sources: []ulid.ULID{ulid.MustNew(1, nil), ulid.MustNew(2, nil)},
+						},
+					},
+					Thanos: metadata.Thanos{
+						Version: 1,
+						Labels:  map[string]string{"a": "1"},
+						Downsample: metadata.ThanosDownsample{
+							Resolution: downsample.ResLevel1,
+						},
+					},
+				},
+				{
+					BlockMeta: tsdb.BlockMeta{
+						ULID:    ulid.MustNew(3, nil),
+						MinTime: 0,
+						MaxTime: downsample.DownsampleRange0,
+						Compaction: tsdb.BlockMetaCompaction{
+							Sources: []ulid.ULID{ulid.MustNew(4, nil), ulid.MustNew(5, nil)},
+						},
+					},
+					Thanos: metadata.Thanos{
+						Version: 1,
+						Labels:  map[string]string{"b": "2"},
+						Downsample: metadata.ThanosDownsample{
+							Resolution: downsample.ResLevel0,
+						},
+					},
+				},
+			},
+			expected: map[string]float64{
+				keys[0]: 0.0,
+				keys[1]: 0.0,
+				keys[2]: 0.0,
+			},
+		},
+	} {
+		if ok := t.Run(tcase.testName, func(t *testing.T) {
+			blocks := make(map[ulid.ULID]*metadata.Meta, len(tcase.input))
+			for _, meta := range tcase.input {
+				blocks[meta.ULID] = meta
+			}
+			groups, err := grouper.Groups(blocks)
+			if err != nil {
+				level.Warn(logger).Log("msg", "unable to form groups")
+			}
```

Comment:
No need to log error. Please use `testutil.Ok` here

Reply:
Thanks, I'm using `testutil.Ok` now for error handling.